rust/src/sip/sip.rs

```diff
@@ -112,15 +112,6 @@ impl SIPState {
         }
     }
 
-    fn build_tx_request(&mut self, input: &[u8], request: Request) {
```

Comment:
You could keep this function (to avoid some code duplication) and simply return the tx_id.

Reply:
I would like to but borrow checker does not like that

Reply:
Why? Do you think the following snippet fails because of the borrow checker?

```
let tx_id = self.build_tx_request(input, request);
sip_frames_ts(flow, &stream_slice, &request, tx_id);
```

Reply:
this failed because `request` cannot be borrowed after having been moved (into the transaction) by `build_tx_request`